packages/oauth/src/index.spec.js

```diff
@@ -0,0 +1,357 @@
+          res.send('failure');
+        },
+      }
+      const installer = new InstallProvider({clientId, clientSecret, stateSecret, installationStore, logger: noopLogger});
+      await installer.handleCallback(req, res, callbackOptions);
+
+      assert.isTrue(sent);
+    });
+    it('should call the success callback for a v2 url', async () => {
+      let sent = false;
+      const installUrlOptions = { scopes: [ 'channels:read' ] };
+      const res = { send: () => { sent = true; } };
+      const callbackOptions = {
+        success: async (installation, installOptions, req, res) => {
+          assert.equal(installOptions.scopes[0], installUrlOptions.scopes[0]);
+          res.send('successful!');
+        },
+        failure: async (error, installOptions , req, res) => {
+          assert.fail(error.message);
+          res.send('failure');
+        },
+      }
+      
+      const installer = new InstallProvider({clientId, clientSecret, stateSecret, installationStore});
+      const url = await installer.generateInstallUrl(installUrlOptions);
+      const req = { url };
+      await installer.handleCallback(req, res, callbackOptions);
+      assert.isTrue(sent);
+
+      try {
+        // do a authorize check
+        const AuthResult = await installer.authorize({teamId:'fake-v2-team-id'});
+        assert.equal(AuthResult.userToken, 'userAccessToken');
+        assert.equal(AuthResult.botToken, 'botToken');
+        assert.equal(AuthResult.botId, '');
+        assert.equal(AuthResult.botUserId, 'botUserId');
+      } catch (error) {
+        assert.fail(error.message);
+      }
+    });
+    it('should call the success callback for a v1 url', async () => {
+      let sent = false;
+      const installUrlOptions = { scopes: [ 'bot' ] };
+      const res = { send: () => { sent = true; } };
+      const callbackOptions = {
+        success: async (installation, installOptions, req, res) => {
+          assert.equal(installOptions.scopes[0], installUrlOptions.scopes[0]);
+          res.send('successful!');
+        },
+        failure: async (error, installOptions , req, res) => {
+          assert.fail(error.message);
+          res.send('failure');
+        },
+      }
+      
+      const installer = new InstallProvider({clientId, clientSecret, stateSecret, installationStore, authVersion: 'v1'});
+      const url = await installer.generateInstallUrl(installUrlOptions);
+      const req = { url };
+      await installer.handleCallback(req, res, callbackOptions);
+      assert.isTrue(sent);
+
+      try {
+        // do a authorize check
+        const AuthResult = await installer.authorize({teamId:'fake-v1-team-id'});
+        assert.equal(AuthResult.userToken, 'token');
+        assert.equal(AuthResult.botToken, 'botAccessToken');
+        assert.equal(AuthResult.botId, '');
+        assert.equal(AuthResult.botUserId, 'botUserId');
+      } catch (error) {
+        assert.fail(error.message);
+      }
+    });
+  })
+});
```

Comment:
a couple tests for `MemoryInstallationStore` and a couple tests for `ClearStateStore` would be helpful.